jenkins/Jenkinsfile

```diff
@@ -0,0 +1,53 @@
+pipeline {
+  agent none
+  options {
+    disableConcurrentBuilds()
+    buildDiscarder(logRotator(numToKeepStr: '8', daysToKeepStr: '20'))
+    timeout(time: 2, unit: 'HOURS')
+  }
+  stages {
+    stage('main') {
+      agent {
+        dockerfile {
+	  dir 'jenkins'
+	  args '--gpus 1'
+        }
+      }
+      environment {
+        HOME = pwd(tmp:true)
+	OMP_NUM_THREADS = 4
+      }
+      steps {
+	sh 'python3 -c "import torch; torch.cuda.current_device()"'
+	sh 'python3 -m venv --system-site-packages $HOME'
+	sh '''#!/bin/bash -ex
+	  source $HOME/bin/activate
+	  pip3 install .
+      for f in tests/test_*py; do python3 -m pytest $f; done
```

Comment:
probably worth adding a comment here to say why we do it this way instead of the obvious `python3 -m pytest tests`

Reply:
back to `pytest tests`, so this isn't relevant anymore